larpandoracontent/LArMonitoring/TwoViewTransverseTracksValidationTool.h

```diff
@@ -0,0 +1,237 @@
+     *  @param  treeDataBox the box where the truth information will be stored
+     *  @param  clusterName a cluster label, needed for TTree branch naming
+     */
+    void CollectTruthInformation(const pandora::Cluster *const pCluster, TreeDataBox &treeDataBox, std::string clusterName);
+
+    /**
+     *  @brief  Convenience function for collecting and registering charge profile information with the pandora monitoring api for writing to a TTree 
+     *
+     *  @param  discreteProbabilityVector the charge profile whose information needs to be stored 
+     *  @param  treeDataBox the box where the charge profile information will be stored
+     *  @param  clusterName a cluster label, needed for TTree branch naming
+     */
+    void CollectChargeProfileInformation(const DiscreteProbabilityVector &discreteProbabilityVector, TreeDataBox &treeDataBox, std::string clusterName);
+
+    /**
+     *  @brief  Check whether two clusters were created by the same true particle 
+     *
+     * @param  pCluster1 the first cluster
+     * @param  pCluster2 the second cluster
+     */
+    bool IsSameMCParticle(const pandora::Cluster *const pCluster1, const pandora::Cluster *const pCluster2);
+
+    pandora::StatusCode ReadSettings(const pandora::TiXmlHandle xmlHandle);
+
+    std::string m_treeName;       ///< name of output tree
+    std::string m_outputFileName; ///< name of output file
+};
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+template <typename T>
+void TwoViewTransverseTracksValidationTool::AbstractDatum::emplace_back(const T &t)
+{
+    emplace_back_impl(std::any(t));
+    return;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+template <typename T>
+TwoViewTransverseTracksValidationTool::TreeDatum<T>::TreeDatum(
+    const T &t, const TwoViewTransverseTracksValidationTool *const pTool, const std::string datumName) :
+    m_Datum(t)
+{
+    pTool->RegisterTreeDatum(m_Datum, datumName);
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+template <typename T>
+void TwoViewTransverseTracksValidationTool::TreeDatum<T>::emplace_back_impl(const std::any &value)
+{
+    if constexpr (is_std_vector<T>::value)
+    {
+        m_Datum.emplace_back((std::any_cast<typename T::value_type>(value)));
+    }
+    else
+        throw pandora::StatusCodeException(pandora::STATUS_CODE_NOT_ALLOWED);
+    return;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+template <typename T>
+void TwoViewTransverseTracksValidationTool::StoreAndRegisterDatum(const T &t, const std::string datumName, TreeDataBox &treeDataBox)
```

Comment:
Called store and register, but doesn't appear to register

Reply:
The registering happens in the `TreeDatum` that's constructed as it is `emplace_back`'d onto the map.  From an interface point of view, storing and registering does happen within that function call but the registering is outsourced to a constructor call, where this happens:
`pTool->RegisterTreeDatum(m_Datum, datumName);`